File: control-flow/Q117/main.go
package main

import "fmt"

func main() {
	var (
		n    int
		sign int = -1
		sum  int
	)
	fmt.Print("Enter n: ")
	fmt.Scan(&n)
	for i := 1; i <= n; i++ {
		if sign == 1 && i != n {
			fmt.Printf("%d + ", i)
		} else if i != n {
			fmt.Printf("%d - ", i)
		} else {
			fmt.Printf("%d = ", i)
		}
		sum += sign * i
		sign *= -1
	}
	fmt.Printf("%d", sum)
}

File: control-flow/Q117/main.rs
use std::io::{self, Read};

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).expect("Failed to read input");
    let n: i32 = input.split_whitespace().next().unwrap().parse().expect("Not a valid number");

    println!("Enter n:");

    let mut sign: i32 = -1;
    let mut sum: i32 = 0;

    for i in 1..=n {
        if sign == 1 && i != n {
            print!("{} + ", i);
        } else if i != n {
            print!("{} - ", i);
        } else {
            print!("{} = ", i);
        }
        sum += sign * i;
        sign *= -1;
    }
    print!("{}", sum);
}